python/llm/src/ipex_llm/transformers/lookup.py

```diff
@@ -56,9 +59,23 @@ def generate(
 ):
     lookahead = kwargs.pop("lookahead", None)
     perf_mode = os.environ.get("IPEX_LLM_PERFORMANCE_MODE", None)
+    use_update_candidate_strategy = True
     if perf_mode == "1" and lookahead is None:
-        lookahead = 2  # default to 2 now
-    if lookahead:
+        if inputs is not None:
+            if inputs.shape[1] >= PERFORMANCE_MODE_LOOKUP_INPUT_THRESHOLD:
+                lookahead = 2  # default to 2 now
+            else:
+                lookahead = 0
+                use_update_candidate_strategy = False
+        else:
+            inputs_embeds = kwargs.get("inputs_embeds", None)
+            if inputs_embeds is not None:
+                if inputs_embeds.shape[1] >= PERFORMANCE_MODE_LOOKUP_INPUT_THRESHOLD:
+                    lookahead = 2  # default to 2 now
+                else:
+                    lookahead = 0
```

Comment:
We set `lookahead=0` when meeting threshold for performance mode benchmarking purposes.

Reply:
May have some further verification whether this way will bring too much overhead.